firestore/api/SolutionCounter/Program.cs

```diff
@@ -0,0 +1,209 @@
+﻿/*
+ * Copyright (c) 2019 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using Google.Cloud.Firestore;
+
+namespace GoogleCloudSamples
+{
+    // Counter is a collection of documents (shards)
+    // to realize counter with high frequency.
+
+    // [START counter_classes]
+
+    // counters/${ID}
+    /// <summary>
+    /// Defines the <see cref="Counter" />
+    /// </summary>
+    [FirestoreData]
+    public class Counter
+    {
+        /// <summary>
+        /// Gets or sets the NumShards
+        /// </summary>
+        [FirestoreProperty]
+        public int NumShards { get; set; }
+    }
+
+    // Shard is a single counter, which is used in a group
+    // of other shards within Counter.
+
+    // counters/${ID}/shards/${NUM}
+    /// <summary>
+    /// Defines the <see cref="Shard" />
+    /// </summary>
+    [FirestoreData]
+    public class Shard
+    {
+        /// <summary>
+        /// Gets or sets the Count
+        /// </summary>
+        [FirestoreProperty]
+        public int Count { get; set; }
+    }
+
+    // [END fs_counter_classes]
+
+    /// <summary>
+    /// Defines the <see cref="DistributedCounter" />
+    /// </summary>
+    public class DistributedCounter
+    { 
+        public static string Usage = @"Usage:
+C:\> dotnet run command YOUR_PROJECT_ID
+
+Where command is one of
+    run-distributed-counter
+";
+        /// <summary>
+        /// Gets or sets the FireStoreDb
+        /// </summary>
+        public static FirestoreDb FirestoreDb { get; set; }
+
+        /// <summary>
+        /// Gets or sets the ShardsCounter
+        /// </summary>
+        public static Counter ShardsCounter { get; set; }
+
+        /// <summary>
+        /// The InitFireStoreDb
+        /// </summary>
+        public static void InitFirestoreDb(string project)
+        {
+            var projectId = project;
+            if (string.IsNullOrEmpty(projectId))
+                throw new ArgumentException("ArgumentException: failed read Firestore Project ID");
```

Comment:
No need to add the text "ArgumentException", that's already clear thanks to the exception type.
Also please include the parameter name.
The message "failed to read..." is misleading, nothing has been read, the method just received a null or empty string as a parameter. You should use the ArgumentNullException which is more suitable and then you can skip the message entirely.